docs/src/conventions.md

```diff
@@ -0,0 +1,123 @@
+# Coding Conventions
+
+## Formatting
+
+We automate most of our formatting preferences. Our CI will run format checkers for PRs and pushes.
+These checks are required for merging any PR.
+
+For Rust, we use [rustfmt](https://github.com/rust-lang/rustfmt)
+which is configured via the [rustfmt.toml](https://github.com/model-checking/kani/blob/main/rustfmt.toml) file.
+We are also in the process of enabling `clippy`.
+Because of that, we still have a couple of lints disabled (see [.cargo/config](https://github.com/model-checking/kani/blob/main/.cargo/config.toml) for the updated list).
+
+We also have a bit of C and Python code in our repository.
+For C we use `clang-format` and for Python scripts we use `autopep8`.
+See [.clang-format](https://githubcom/model-checking/kani/blob/main/.clang-format)
+and [pyproject.toml](https://github.com/model-checking/kani/blob/main/scripts/pyproject.toml)
+for their configuration.
+
+
+### Exceptions
+
+We recognize that in some cases, the formatting and lints automation may not be applicable to a specific code.
+In those cases, we usually prefer explicitly allowing exceptions by locally disabling the check.
+E.g., use `#[allow]` annotation instead of disabling a link on a crate or project level.
+
+### Copyright notice
+
+All source code files begin with a copyright and license notice. If this is a new file, please add the following notice:
+
+```rust
+// Copyright Kani Contributors
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+```
+
+When modifying a file from another project, please keep their headers as is and append the following notice after them:
+
+```rust
+// ... existing licensing headers ...
+
+// Modifications Copyright Kani Contributors
+// See GitHub history for details.
+```
+
+Note: The comment escape characters will depend on the type of file you are working with. E.g.: For rust start the 
+header with `//`, but for python start with `#`.
+
+We also have automated checks for the copyright notice.
+There are a few file types where this rule doesn't apply.
+You can see that list in the [copyright-exclude](
+https://github.com/model-checking/kani/blob/main/scripts/ci/copyright-exclude) file.
+
+
+## Code for soundness
+
+We are developing Kani to provide assurance that critical Rust components are verifiably free of certain classes of
+security and correctness issues.
+Thus, it is critical that we provide a verification tool that is sound.
+For the class of errors that Kani can verify, we should not produce a “No Error” result if there was in fact an
```

Comment:
Nit: use normal double quotes around "No Error" instead of the UTF-8 left/right double quotes.